auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneActivity.java

```diff
@@ -14,401 +14,204 @@
-
-    private static final String KEY_VERIFICATION_PHONE = "KEY_VERIFICATION_PHONE";
-    private static final String KEY_STATE = "KEY_STATE";
-
-    private AlertDialog mAlertDialog;
-    @VisibleForTesting CompletableProgressDialog mProgressDialog;
-    private Handler mHandler;
-    private String mPhoneNumber;
-    private String mVerificationId;
-    private Boolean mIsDestroyed = false;
-    private PhoneAuthProvider.ForceResendingToken mForceResendingToken;
-    private VerificationState mVerificationState;
-
-    public static Intent createIntent(Context context, FlowParameters flowParams, Bundle params) {
-        return createBaseIntent(context, PhoneActivity.class, flowParams)
-                .putExtra(ExtraConstants.PARAMS, params);
+    public static Intent createIntent(Context context, FlowParameters params, Bundle args) {
+        return createBaseIntent(context, PhoneActivity.class, params)
+                .putExtra(ExtraConstants.PARAMS, args);
     }
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fui_activity_register_phone);
 
-        mHandler = new Handler();
-        mVerificationState = VerificationState.VERIFICATION_NOT_STARTED;
-        if (savedInstanceState != null && !savedInstanceState.isEmpty()) {
-            mPhoneNumber = savedInstanceState.getString(KEY_VERIFICATION_PHONE);
+        final PhoneProviderResponseHandler handler =
+                ViewModelProviders.of(this).get(PhoneProviderResponseHandler.class);
+        handler.init(getFlowParams());
+        handler.getOperation().observe(this, new ResourceObserver<IdpResponse>(
+                this, R.string.fui_progress_dialog_signing_in) {
+            @Override
+            protected void onSuccess(@NonNull IdpResponse response) {
+                startSaveCredentials(handler.getCurrentUser(), response, null);
+            }
+
+            @Override
+            protected void onFailure(@NonNull Exception e) {
+                handleError(e);
+            }
+        });
 
-            if (savedInstanceState.getSerializable(KEY_STATE) != null) {
-                mVerificationState = (VerificationState) savedInstanceState.getSerializable(
-                        KEY_STATE);
+        final PhoneNumberVerificationHandler phoneVerifier =
+                ViewModelProviders.of(this).get(PhoneNumberVerificationHandler.class);
+        phoneVerifier.init(getFlowParams());
+        phoneVerifier.getOperation().observe(this, new ResourceObserver<PhoneVerification>(
+                this, R.string.fui_verifying) {
+            @Override
+            protected void onSuccess(@NonNull PhoneVerification verification) {
+                if (verification.isAutoVerified()) {
+                    Toast.makeText(
+                            PhoneActivity.this, R.string.fui_verified, Toast.LENGTH_LONG).show();
```

Comment:
Does this address your auto verified concerns?

Reply:
Oh also I just went through and tried this, I think we should extend the `fui_verified` method to be something like "Phone number automatically verified"